### PR TITLE
Major bug fixes

### PR DESCRIPTION
**Pull request by Nirmalya24**
Co-Authored with Jen and Troy
- Fixed `add/item` and `add/section` routes
  -  Fix the underlying function calls. Updated the MongoDB/Mongoose Query to use `aggregate` 
- Updated function to be asynchronous in nature

### CODE REVIEW

File: model/ItemModel.ts [hunk 1]
```diff
@@ -63,13 +63,32 @@ class ItemModel {
...
             return itemArray;
+
         });
+
+    }
+    */
+
+     public getItem(filter:object): any {
+        return new Promise((resolve, reject) => {
+            const query = this.model.findOne(filter);
```
Comment (jen-chieh) [inline model/ItemModel.ts:81]: This part should type of Promise

File: createDB/createSampleItems.js [hunk 2]
```diff
@@ -197,18 +197,19 @@ const itemsData = [
...
     },
     {
-        "itemName": "",
-        "itemDescription": "",
-        "itemPrice": ,
+        "itemID": "b061f2c6-e85c-11ed-a05b-0242ac120003",
+        "itemName": "Chocolate Cake",
+        "itemDescription": "Chocolate flavored cake",
+        "itemPrice": 8.99,
         "itemImg": "https://example.com/images/.jpg",
```
Comment (goyaladit) [inline createDB/createSampleItems.js:204]:
```suggestion
        "itemImg": "https://example.com/images/chocolatecake.jpg",
```

File: createDB/createSampleItems.js [hunk 3]
```diff
@@ -197,18 +197,19 @@ const itemsData = [
...
         "menusID": [""]
     },
     {
-        "itemName": "",
-        "itemDescription": "",
-        "itemPrice": ,
+        "itemName": "Cheesecake",
+        "itemDescription": "Cheesecake with strawberry topping",
+        "itemPrice": 6.99,
         "itemImg": "https://example.com/images/.jpg",
```
Comment (goyaladit) [inline createDB/createSampleItems.js:213]:
```suggestion
        "itemImg": "https://example.com/images/cheesecake.jpg",
```

File: model/ItemModel.ts (hunk 1 above)
Comment (Nirmalya24) [reply model/ItemModel.ts:82]: Done

**Review by jen-chieh — APPROVED** [review]
Good

File: model/MenuModel.ts [hunk 4]
```diff
@@ -24,8 +25,8 @@ class MenuModel {
                 menuItems: [String],
                 menuSections: Object,
                 menuDescription: String,
-                menuStartTime: Date,
-                menuEndTime: Date
+                menuStartTime: String,
+                menuEndTime: String
```
Comment (goyaladit) [inline model/MenuModel.ts:29]:
```suggestion
                menuStartTime: string,
                menuEndTime: string
```

to be consistent with menuStart and menuEnd in other file

File: model/MenuModel.ts [hunk 5]
```diff
@@ -171,7 +207,7 @@ class MenuModel {
         });
     }
 
-    public updateMenuTime(response: any, filter: object, startTime: Date, endTime: Date): any {
+    public updateMenuTime(response: any, filter: object, startTime: String, endTime: String): any {
```
Comment (goyaladit) [inline model/MenuModel.ts:210]:
```suggestion
    public updateMenuTime(response: any, filter: object, startTime: string, endTime: string): any {
```

**Review by goyaladit — COMMENTED** [review]
very minor consistency issues

**Pull request merged**